src/components/Notifications/index.tsx

```diff
@@ -54,7 +54,7 @@ const generateNotificationId = () => {
 
 const getTransformedOptions = (
 	options: React.ReactNode | NotificationOptions,
-): NotificationOptions => {
```

Comment:
Why did we remove this return typing?

Reply:
@myarmolinsky B/c the inferred return type has the `id` as non-nullable, while NotificationOptions have it as nullable, which made ts complain during an assignment later in the file.

Reply:
@thgreasi Yes I saw that, but `getTransformedOptions` sets some default values. It's possible for them to be overriden if the user explicitly passes undefined for them but since the values are required it makes sense for us to just use `??` to ensure the value is always defined even if the user explicitly passes `undefined`

Reply:
Ok ok fine....

Reply:
really strange one btw, TS is not considering the `id: undefined` case...  checking TS issues....